### DISCUSSION TITLE
Melléklapok mezői

### DISCUSSION THREAD

**Discussion by Macskafarka** [nav-gov-hu/eVAT#84] [Kérdések és válaszok - Gépi interfész]
Az eÁFA_M2M_rendszer_interfesz_specifikacio_v0.5.pdf 27. oldalán a melléklapok szerkezetének nincs megadva a pontos leírása, hanem csak egy visszautalás az ÁNYK nyomtatványra: "A melléklapok szerkezete az ÁNYK XML formátumot követi, egy kulcs érték párral (sheetFieldName + sheetFieldValue) leírhatók."  Még a nyomtatvány sincs konkrétan megnevezve. Nekünk kellene kitalálni azt, hogy vajon a 2308-as, vagy a 2365-ös bevallásra gondoltak-e? a 2465-re nyilván nem gondolhat a specifikáció, mert a specifikáció elkészültekor még nem jelent meg. Amúgy már 15 napja megjelent a NAV oldalain, hogy "Elindult az eÁFA". Ehhez képest hiányoznak a pontos meződefiniciók, és a fejlesztőkre van bízva, hogy a jövőben majdan esetlegesen megjelenő 2465 bevallásból vegyék ki a specifikáció szerint "kulcs"-nak nevezett valamit, amit az ÁNYK nyomtatványon nem kulcsnak hívnak. Ott ilyenek vannak: "Közös azonosító"; "Import azonosító"; és "Belső azonosító". Aki M2M eÁFA bevallást fejleszt, annak minden xx65 ÁNYK nyomtatvány verzió (nemcsak év elején, hanem év közben is) megjelenéskor ellenőriznie kell, hogy megváltoztak-e a "kulcsok", akármit is jelentsen a kulcs kifejezés? Sokszor a nyomtatvány verziónál csak annyi van közzétéve a NAV letöltő oldalán: "Tecnikai módosítás", amibe bőven belefér pár mező átnevezése.  Hogyan tudná kiváltani az eÁFA az ÁNYK bevallást, ha annak mezőszintű ismerete kell hozzá? Lehetne kérni olyan specifikációt, amelyben konkrétumok szerepelnek?

**Comment by bmatyi94** [comment]
ehhez én is csatlakoznék. A terv hogy kiváltsuk az ÁNYK bevallást hosszútávon. Valamint, ha beadjuk az áfa analitika xml-t, a NAV visszaadja a bevfeld verziót. Viszont a melléklapok nincsenek összehangban ezzel, hiszen már eleve az ÁNYK azonosítókat kell használni. 07-08 as lapokhoz kapcsolódó adatot (vámtarifaszám, mennyiség, nettó és áfa összege) nem kötelező ágként a számlaadatokhoz lenne jó hozzáadni az xml-ágban. Ebből pedig az Adóhatóság össze tudja állítani a melléklapot. Ez csak egy példa, de a többi melléklapnál is hasonló logika alkalmazható.

**Reply by omachtandras** [reply]
Méretben is kisebb lenne az átadandó állomány. Tudom, hogy ez miatt nem szabad(na) aggódni, de azért nálunk vannak ügyfelek több százezer számlával, meg lesznek olyanok is biztos, akiknek millió számra van nyugtájuk...
(A magam részéről sokkal jobbank tartottam volna, ha számlánként lehet akár jelezni, ha valahol nem értek egyet a beajánlott adókódolással, mert vevő számla oldalon azért jó eséllyel a NAV el fogja találni, hogy minek kellene és milyen kódolással a bevallásba kerülni, ezt az adatot még egyszer felküldeni egybe csomagolva izgalmas lesz...)

**Comment by Macskafarka** [comment]
Az, hogy a VAT_SHEET_7 -ben a mennyiséget Kg-ban kell megadni, az is kizárólag a 2365 bevallás súgójából hámozható ki.

**Comment by renced42** [comment]
"Az eÁFA_M2M_rendszer_interfesz_specifikacio_v0.5.pdf 27. oldalán a melléklapok szerkezetének nincs megadva a pontos leírása,"
Kedves @Macskafarka 

Köszi az észrevételt kiegészítjük a dokumentációt.

**Comment by devbandras** [comment]
Sziasztok!

Ebben történt valami változás? A mezőnevekre a jövőben is ilyen azonosítóval kell majd hivatkozni ami az ANYK-ban van?  Most elő kellet vennem egy régebbi fejlesztést, hogy az új oszlopokat ki tudjuk tölteni (Levonásra helyezett adó, 5%, 18%, 27%). A 02-es lapon elindul a 29632-es mezőkóddal ami egyesével növekszik a megfelelő oszlopoknál, majd a 7-sornál történik egy törés. (29655 után 29680 következik). Hasonló eset van a 13. sornál is. A 02-K lapon is van hasonló eset. A mai világba ezeket a sorokat nem egy "
beszédes" listába tenni? 

![anyk_2565M](https://github.com/user-attachments/assets/4ea3d877-6106-4250-b744-182c54473678)

**Comment by NTCA-tax** [comment]
A 2.0-ás XSD vezeti ki teljes mértékben az ÁNYK mezőkódokat. Ezt követően egyszerűbb, könnyebben átláthatóbb adatstruktúrát kell kitölteni.
